libdnf5/base/transaction_module.cpp

```diff
@@ -34,7 +34,8 @@ class TransactionModule::Impl {
     Action action;
     Reason reason;
     std::string module_name;
-    std::string module_stream;
+    std::string source_stream;
+    std::string target_stream;
```

Comment:
Looking at https://github.com/fedora-modularity/libmodulemd/blob/main/yaml_specs/modulemd_obsoletes_v1.yaml I think it would be possible that a `TransactionModule` should change to a different `module:stream` (not only a stream).

What about to use similar approach like `TransactionPackage`, add something like `get_replaces()` or `get_replaced_by()`?

Reply:
Thanks for the suggestion! I am not quite sure how the obsoletes will be implemented, but I modified the `TransactionModule` similarly to the `TransactionPackage`. The `get_replaced_by()` is not needed for stream switching, but I added it, too, for later use for obsoletes.